drivers/serial/uart_nrfx_uarte.c

```diff
@@ -20,19 +41,70 @@
      defined(CONFIG_UART_2_INTERRUPT_DRIVEN)) || \
     (defined(CONFIG_UART_3_NRF_UARTE) &&         \
      defined(CONFIG_UART_3_INTERRUPT_DRIVEN))
-	#define UARTE_INTERRUPT_DRIVEN	(1u)
+	#define UARTE_INTERRUPT_DRIVEN	1
 #endif
 
+#define MS_PER_TICK (CONFIG_SYS_CLOCK_TICKS_PER_SEC <= 1000 ?	\
+		     1000 / CONFIG_SYS_CLOCK_TICKS_PER_SEC : 1)
```

Comment:
This integer division should be rather rounded up. You can use e.g. `NRFX_CEIL_DIV` for this.
But actually, do we still need this value of `MS_PER_TICK`? It is used only in calculation of `rx_timeout_slab`. Won't the effect be the same if you use just `1` there?